libtenzir/include/tenzir/type.hpp

```diff
@@ -1340,6 +1340,35 @@ class record_type final : public stateful_type_base {
   friend record_type flatten(const record_type& type) noexcept;
 };
 
+// -- blob_type --------------------------------------------------------------
+
+/// An sequence of bytes.
+/// @relates type
+class blob_type final {
+public:
+  /// Returns the type index.
+  static constexpr uint8_t type_index = 16;
+
+  // TODO: Or LargeBinaryType?
+  /// The corresponding Arrow DataType.
+  using arrow_type = arrow::BinaryType;
+
+  /// Returns a view of the underlying binary representation.
+  friend std::span<const std::byte> as_bytes(const blob_type&) noexcept;
+
+  // TODO: Or chunk_ptr?
```

Comment:
The advantage of using `chunk_ptr` for blobs is that we don't need to copy (as we can slice with a shared lifetime), and that it can be both view and data at the same time. However, I think it has the precondition to remove `chunk_ptr` and to always make `chunk` have shared pointer semantics, and I don't think that's a good use of our time right now.

Reply:
Alright, then let's leave it as it is for now. We can always change this later.